face detection

Jdavies08 opened the pull request:
code is now able to detect faces and draw a box around them

Jdavies08 committed:
face detection

code is now able to detect faces and draw a box around them

HarrisonHG commented on main.py:
```diff
@@ -1,10 +1,20 @@
 	
+	
+	
+	
```
Bit too much white space - keep it to 1 empty line if whitespace is needed

HarrisonHG commented on main.py:
```diff
@@ -1,10 +1,20 @@
+	#detect the faces
+	for top,right,bottom,left in face_recognition.face_locations(frame):
+		
+		#draw rectangle around faces
```
Comments around each line can pad the file too much.
Use comments to explain "why" you did something if it doesn't feel obvious.
The line below this is clear enough to not need a comment.

HarrisonHG commented on main.py:
```diff
@@ -1,10 +1,20 @@
-import cv2
+import cv2, face_recognition
 #get webcam
 cap = cv2.VideoCapture(0)
```
Using zero here is a [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)#:~:text=Magic%20numbers%20are%20common%20in%20at%20program%20run-time.).

Use a constant at the top of the file, assign 0 to it, then use the constant value here.
This also futureproofs the code against any necessary changes in the future, such as selecting a camera index if there are several.

HarrisonHG commented on main.py:
```diff
@@ -1,10 +1,20 @@
+	
+	#shows the image
 	cv2.imshow("window",frame)
+	#quit if q is pressed
```
unnecessary comment

HarrisonHG commented on main.py:
```diff
@@ -1,10 +1,20 @@
+	
+	
+	
+	#shows the image
```
unnecessary comment

HarrisonHG commented on main.py:
```diff
@@ -1,10 +1,20 @@
-import cv2
+import cv2, face_recognition
 #get webcam
```
a [doc string](https://peps.python.org/pep-0257/) for the file as a whole would be good. What's the point of this script? What are you going to achieve with it?

HarrisonHG reviewed (changes requested):
- I was warned when running the script that you **might** (unsure) also need to add "face_recognition_models" to the requirements.txt file above face-recognition.
- Have you run this through pylint? Aim for 10/10!
- Create a file called README.md, in this file, describe your goals for the project, how you run the script (such as using the command `python main.py`), any requirements needed before running the script (such as installing python, pip and how to use pip with the requirements file). This is the developer's first point of documentation, so anyone that isn't in the room developing this with you has the guidance needed to run this even if they'd never touched python before.

Jdavies08 committed:
Improved for clearer running

added pylint to requirements.txt ensured everything had 10/10 scores on pylint and added README.md to give an guide to run the code which will be updated as it is improved

Jdavies08 committed:
Fixed Readme
